### PR TITLE
Bump moto-ext to 4.2.7.post1

### PR DESCRIPTION
**Pull request by viren-nadkarni**
This is the final moto-ext bump ahead of LocalStack v3.0.0 release.

Brings in:
- https://github.com/getmoto/moto/pull/7003
- https://github.com/getmoto/moto/pull/7004

### CODE REVIEW

File: tests/aws/services/sns/test_sns.py [hunk 1]
```diff
@@ -3302,48 +3302,6 @@ def check_message():
...
-                PlatformApplicationArn=platform_arn,
-                Token=token,
-                CustomUserData="different-user-data",
-            )
-        assert e.value.response["Error"]["Code"] == "InvalidParameter"
-        assert (
-            e.value.response["Error"]["Message"]
-            == f"Endpoint {endpoint_arn} already exists with the same Token, but different attributes."
-        )
-
```
Comment (viren-nadkarni) [inline tests/aws/services/sns/test_sns.py:3346]: This behaviour is now correctly tested upstream in Moto: https://github.com/getmoto/moto/pull/6953

File: localstack/services/sns/provider.py [hunk 2]
```diff
@@ -342,21 +342,7 @@ def create_platform_endpoint(
...
-                for e in moto_sns_backend.platform_endpoints.values():
-                    if e.token == token:
-                        if custom_user_data and custom_user_data != e.custom_user_data:
-                            raise InvalidParameterException(
-                                f"Endpoint {e.arn} already exists with the same Token, but different attributes."
-                            )
-                        else:
-                            return CreateEndpointResponse(EndpointArn=e.arn)
-            raise
-        return result
```
Comment (viren-nadkarni) [inline localstack/services/sns/provider.py:359]: This behaviour is now correctly implemented and tested upstream in Moto: https://github.com/getmoto/moto/pull/6953

File: tests/aws/services/sns/test_sns.py [hunk 3]
```diff
@@ -3302,48 +3302,6 @@ def check_message():
 
         retry(check_message, retries=PUBLICATION_RETRIES, sleep=PUBLICATION_TIMEOUT)
 
-    @markers.aws.needs_fixing
-    # AWS validating this is hard because we need real credentials for a GCM/Apple mobile app
-    # Error responses are from reported https://github.com/spulec/moto/issues/2333
-    def test_create_platform_endpoint_check_idempotency(
```
Comment (bentsku) [inline tests/aws/services/sns/test_sns.py:3308]: I wouldn't delete the test, even if validated in Moto. Does it still pass?

**Review by bentsku — COMMENTED** [review]
LGTM for the SNS changes, but do we really need to delete the test? I'd like to keep, I'm planning on AWS validating them at some point.

**Review by alexrashed — APPROVED** [review]
Lookin' good! Thanks for keeping the dependency up to date! 🦸‍♀️

File: tests/aws/services/sns/test_sns.py (hunk 3 above)
Comment (viren-nadkarni) [reply tests/aws/services/sns/test_sns.py:3308]: Fair point about AWS validating the test. It asserts wrong behaviour so skipped for now. 

Added it back with some pointers :+1:

Comment (bentsku) [reply tests/aws/services/sns/test_sns.py:3308]: Awesome, thanks! Yeah, Moto says the test has been AWS now, so I guess we can trust them, maybe he has some way of testing against AWS with an Apple developer account and fake app! We will need to get to it at some point. Thanks for putting it back!

**Review by bentsku — APPROVED** [review]

**Pull request merged**